arbor/backends/gpu/matrix_fine.cu

```diff
@@ -200,23 +212,29 @@ void solve_matrix_fine(
 
     // take great care with loop limits decrementing unsigned counter l
     for (unsigned l=num_levels-1; l>0; --l) {
-        const auto& lvl = block_levels[l-1];
+        const auto& lvl_meta = block_levels_meta[l-1];
```

Comment:
This definition shadows the one on line 175.

The `next_lvel_meta` variable is used only in one place, on line 224, so you might as well inline it.

Reply:
Will change the previous one to `last_lvl_meta` and inline `next_lvl_meta`